roachtest: add Mocha test parser for node-postgres tests

spilchen opened the pull request:
This commit introduces Mocha test output parsing for the node-postgres roachtest. The parser handles hierarchical test suites (suite => subsuite => test) and properly categorizes test results against blocklists and ignorelists.

With this ignorelist in place, this also ignores a test flake as outlined here: https://github.com/cockroachdb/cockroach/issues/152728#issuecomment-3245640238

Fixes #152728

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154520)
<!-- Reviewable:end -->

rafiss reviewed (approved):
this should save a lot of toil!

did you get to try a real execution by running `roachtest` manually?

spilchen committed:
roachtest: add Mocha test parser for node-postgres tests

This commit introduces Mocha test output parsing for the node-postgres
roachtest. The parser handles hierarchical test suites
(suite => subsuite => test) and properly categorizes test results against
blocklists and ignorelists.

With this ignorelist in place, this also ignores a test flake as outlined here:
https://github.com/cockroachdb/cockroach/issues/152728#issuecomment-3245640238

Fixes #152728

Release note: None
Epic: None

spilchen reviewed (commented):
It doesn't take long to run, so once I had all of the changes sorted out I ran it in a tight loop. I got about 12 successes in a row before cancelling the loop.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154520#-:-OaVximu1ueRJfig5y6h:b-dphgxe)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @spilchen)



<!-- Sent from Reviewable.io -->
